ros2topic/ros2topic/verb/pub.py

```diff
@@ -43,16 +43,25 @@ def add_arguments(self, parser, cli_name):
             help='Values to fill the message with in YAML format ' +
                  '(e.g. "data: Hello World"), ' +
                  'otherwise the message will be published with default values')
+        parser.add_argument(
+            '-r', '--rate', metavar='N', type=float, default=1.0,
+            help='Publishing rate (hz) default to 1')
+        parser.add_argument(
+            '-1', '--once', action='store_true',
+            help='Publish one message and exit')
 
     def main(self, *, args):
         return main(args)
 
 
 def main(args):
-    return publisher(args.message_type, args.topic_name, args.values)
+    if args.rate <= 0:
+        raise ValueError('rate must be greater than zero')
```

Comment:
I would suggest ti move this into `PubVerb.main`. That way the entity defining the command line arguments is also responsible of checking the sanity of the values. (The same way as it is for the `CallVerb` above.)

Reply:
Does the ValueError result in a stack trace for the user? If so I would recommend to either use argparse to signal the error or raise a RuntimeError (both with the goal to not show a stack trace to the user).

Reply:
Good point, I changed it to a RuntimeError

Reply:
> I would suggest ti move this into PubVerb.main. That way the entity defining the command line arguments is also responsible of checking the sanity of the values. (The same way as it is for the CallVerb above.)

Good point, moved in https://github.com/ros2/ros2cli/pull/66/commits/041b4111122b42a1c5dadebcff5dcca9d91e09b1